app/llama_sensei/backend/qa/generate_answer.py

```diff
@@ -1,31 +1,46 @@
+import time
 from datetime import datetime
 
+import numpy as np
 from datasets import Dataset
+from langchain_community.embeddings import SentenceTransformerEmbeddings
+from langchain_community.utilities import DuckDuckGoSearchAPIWrapper
 from langchain_groq import ChatGroq
+from llama_sensei.backend.add_courses.embedding.get_embedding import Embedder
 from llama_sensei.backend.add_courses.vectordb.document_processor import (
     DocumentProcessor,
 )
 from ragas import evaluate
 from ragas.metrics import faithfulness
+from sklearn.metrics.pairwise import cosine_similarity
 
 MODEL = "llama3-70b-8192"
+# device = "cuda" if torch.cuda.is_available() else "cpu"
+# EMBEDDING_LLM = SentenceTransformer("all-MiniLM-L12-v2", trust_remote_code=True).to(device)
+EMBEDDING_LLM = SentenceTransformerEmbeddings(
```

Comment:
Please change to EMBEDDING_LLM = "all-MiniLM-L12-v2" and initialize the embedder below (inside the GenerateRAGAnswer __init__ function)